apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/RpcRetryConsumer.cs

```diff
@@ -83,7 +83,13 @@ public void Receive(IEnumerable<T> items)
             {
                 _consumer.Receive(items);
             }
-            catch (RpcException e) when (HandleException(items, e)) { }
+            catch (RpcException e)
+            {
+                if (!HandleException(items, e))
+                {
+                    throw;
+                }
+            }
```

Comment:
any exception thrown in an exception filter is swallowed, this makes throwing in the filter impossible.